### PR TITLE
19 us as a user i want to see my friends status online offline playing

### PR DESCRIPTION
**Pull request by cnstll**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [ ] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [x] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: backend/app/src/user/user.service.ts [hunk 1]
```diff
@@ -153,8 +155,21 @@ export class UserService {
...
+          id: userId,
+        },
+        data: {
+          status: UserStatus.OFFLINE,
+        },
+      });
+      return res.status(200).clearCookie('jwtToken', { httpOnly: true }).send();
+    } catch (error) {
+      //UserId was not found
+      return res.status(401).send();
```
Comment (COTHSC) [inline backend/app/src/user/user.service.ts:171]: Are we sure this is the right way to handle this error? shouldn't we clear the cookie regardless of errors  in the db? and is a 401 the right error status to send back?

File: backend/app/src/user/user.gateway.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  @WebSocketServer()
+  server: Server;
+  constructor(private readonly userService: UserService) {}
+
+  @SubscribeMessage('connectUser')
+  async userConnect(
+    @GetCurrentUserId() userId: string,
+    @ConnectedSocket() clientSocket: Socket,
+  ) {
+    await this.userService.updateConnectionStatus(userId, UserStatus.ONLINE);
```
Comment (COTHSC) [inline backend/app/src/user/user.gateway.ts:31]: This return in not error handled, and will emit the same thing wether or not the db write was succesful or not, is that the desired effect?

File: frontend/app/src/components/section-components/friend-list.tsx [hunk 3]
```diff
@@ -1,14 +1,43 @@
...
 import useUserFriends from '../query-hooks/useUserFriends';
 import LoadingSpinner from './loading-spinner';
 import UsersList from './users-list';
 
 function FriendsList() {
   const friends = useUserFriends();
+  const queryClient = useQueryClient();
+  useEffect(() => {
+    socket.on('userDisconnected', async () => {
+      await queryClient.invalidateQueries('friendsList');
```
Comment (COTHSC) [inline frontend/app/src/components/section-components/friend-list.tsx:13]: Why wait?

File: frontend/app/src/components/section-components/friend-list.tsx [hunk 4]
```diff
@@ -1,14 +1,43 @@
...
+    });
+    socket.on('userConnected', async () => {
+      console.log('PROFILE - USER CONNECTED');
```
Comment (COTHSC) [inline frontend/app/src/components/section-components/friend-list.tsx:16]: Do you still need the console log?

File: frontend/app/src/components/section-components/friend-list.tsx (hunk 3 above)
Comment (cnstll) [reply frontend/app/src/components/section-components/friend-list.tsx:13]: promises voided

File: backend/app/src/user/user.service.ts (hunk 1 above)
Comment (cnstll) [reply backend/app/src/user/user.service.ts:171]: No. But are we sure of ? Will the sun rise and shine tomorrow ? ..

File: frontend/app/src/components/section-components/friend-list.tsx (hunk 4 above)
Comment (cnstll) [reply frontend/app/src/components/section-components/friend-list.tsx:16]: removed

**Pull request merged**